src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorExtensions.cs

```diff
@@ -0,0 +1,1754 @@
+        //        var oslice = outputs[i].Slice(outputs[i].Lengths);
+
+        //        nint copiedValues = 0;
+        //        while (copiedValues < totalToCopy)
+        //        {
+        //            SpanHelpers.Memmove(ref Unsafe.Add(ref oslice._reference, SpanHelpers.GetIndex(oIndices, outputs[0].Strides, outputs[0].Lengths)), ref Unsafe.Add(ref islice._reference, SpanHelpers.GetIndex(iIndices, islice.Strides, islice.Lengths)), copyLength);
+        //            SpanHelpers.AdjustIndices((int)axis, 1, ref oIndices, outputs[i].Lengths);
+        //            SpanHelpers.AdjustIndices((int)axis - 1, 1, ref iIndices, input.Lengths);
+        //            copiedValues += copyLength;
+        //        }
+        //    }
+
+        //    return outputs;
+        //}
+        #endregion
+
+        #region SetSlice
+        // REVIEW: WHAT DO WE WANT TO CALL THIS? COPYTO? IT DOES FIT IN WITH THE EXISTING COPY TO CONVENTIONS FOR VECTOR (albeit backwards).
+        public static Tensor<T> SetSlice<T>(this Tensor<T> tensor, Tensor<T> values, params NativeRange[] ranges)
+            where T : IEquatable<T>, IEqualityOperators<T, T, bool>
+        {
+            SpanND<T> srcSpan;
+            if (ranges == Array.Empty<NativeRange>())
+            {
+                if (!tensor.Lengths.SequenceEqual(values.Lengths))
+                    throw new ArgumentException("When no ranges are specified the values tensor must be equal in size as the input tensor.", nameof(values));
+                srcSpan = tensor.AsSpan().Slice(tensor.Lengths);
+            }
+            else
+                srcSpan = tensor.AsSpan().Slice(ranges);
+
+            if (!srcSpan.Lengths.SequenceEqual(values.Lengths))
+                throw new ArgumentException("Provided values must have the same shape as the input tensor.", nameof(values));
+
+            values.AsSpan().CopyTo(srcSpan);
+
+            return tensor;
+        }
+        #endregion
+
+        #region FilteredUpdate
+        // REVIEW: PYTORCH/NUMPY DO THIS.
+        //  t0[t0 < 2] = -1;
+        //  OR SHOULD THIS BE AN OVERLOAD OF FILL THAT TAKES IN A FUNC TO KNOW WHICH ONE TO UPDATE?
+        public static Tensor<T> FilteredUpdate<T>(Tensor<T> left, Tensor<bool> filter, T value)
+            where T : IEquatable<T>, IEqualityOperators<T, T, bool>
+        {
+            if (filter.Lengths.Length != left.Lengths.Length)
+                throw new ArgumentOutOfRangeException(nameof(filter), "Number of dimensions to slice does not equal the number of dimensions in the span");
+
+            var srcSpan = MemoryMarshal.CreateSpan(ref left._values[0], (int)left._linearLength);
+            var filterSpan = MemoryMarshal.CreateSpan(ref filter._values[0], (int)left._linearLength);
+
+            for (int i = 0; i < filterSpan.Length; i++)
+            {
+                if (filterSpan[i])
+                {
+                    srcSpan[i] = value;
```

Comment:
Why is the destination being written to called "src"?

Reply:
Fixed.